src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -206,9 +206,15 @@ public override async Task<DValue<BooleanValue>> RemoveAsync(IEnumerable<Formula
 
                     if (await MatchesAsync(dRecord.Value, recordToRemove, cancellationToken).ConfigureAwait(false))
                     {
-                        found = true;
-
-                        deleteList.Add(item);
+                        if (markedToDeletion.Contains(item))

```

Comment:
>markedToDeletion.Contains(item) [](http://example.com/codeflow?start=28&length=31)

does this have issues with using T as a hash key? (similar to the DValue issues )